src/App/backend/src/Altinn.App.Core/Internal/LocaltestValidation.cs

```diff
@@ -202,4 +243,169 @@ private async Task<VersionResult> Version()
+            attempt++;
+        }
+
+        _logger.LogWarning("Server addresses not available after 10 seconds");
+    }
+
+    private async Task RegisterWithLocaltest(string baseUrl, int port, CancellationToken stoppingToken)
+    {
+        var applicationMetadata = await _appMetadata.GetApplicationMetadata();
+        var appId = applicationMetadata.Id;
+
+        string? hostname = null;
+        if (Environment.GetEnvironmentVariable("DOTNET_RUNNING_IN_CONTAINER") == "true")
+        {
+            // Running inside a container - use container's hostname
+            hostname = Environment.GetEnvironmentVariable("HOSTNAME");
+        }
+
+        var registrationRequest = new
+        {
+            appId = appId,
+            port = port,
+            hostname = hostname,
+        };
+
+        var url = $"{baseUrl}/Home/Localtest/Register";
+
+        int maxRetries = 3;
+        int retryCount = 0;
+
+        while (retryCount < maxRetries && !stoppingToken.IsCancellationRequested)
+        {
+            try
+            {
+                using var client = _httpClientFactory.CreateClient();
+                var response = await client.PostAsJsonAsync(url, registrationRequest, stoppingToken);
+
+                if (response.IsSuccessStatusCode)
+                {
+                    _registeredAppId = appId;
+                    _logger.LogInformation(
+                        "Successfully registered {AppId} with localtest on {Hostname}:{Port}",
+                        appId,
+                        hostname ?? "host.docker.internal",
+                        port
+                    );
+                    return;
+                }
+                else if (response.StatusCode == System.Net.HttpStatusCode.NotFound)
+                {
+                    // 404 means localtest doesn't support registration endpoint
+                    _logger.LogError(
+                        "App is running on port {Port} (not 5005), but localtest doesn't support app registration. "
+                            + "Please upgrade your localtest installation (git pull) to support apps on custom ports.",
+                        port
+                    );
+                    Exit();
+                    return;
+                }
+                else
+                {
+                    var errorContent = await response.Content.ReadAsStringAsync(stoppingToken);
+                    _logger.LogWarning(
+                        "Failed to register app with localtest (attempt {Attempt}/{MaxAttempts}). Status: {StatusCode}, Error: {Error}",
+                        retryCount + 1,
+                        maxRetries,
+                        response.StatusCode,
+                        errorContent
+                    );
+                }
+            }
+            catch (HttpRequestException ex) when (ex.StatusCode == System.Net.HttpStatusCode.NotFound)
```

Comment:
I don't think this exception will be thrown, as you check `response.IsSuccessStatusCode` instead of calling `response.EnsureSuccessStatusCode()`